packages/@react-spectrum/s2/src/Tabs.tsx

```diff
@@ -314,50 +302,344 @@ function TabLine(props: TabLineProps) {
+    isSelected: 'neutral',
+    isHovered: 'neutral-subdued',
+    isDisabled: 'disabled',
+    forcedColors: {
+      isSelected: 'Highlight',
+      isDisabled: 'GrayText'
+    }
+  },
+  borderRadius: 'sm',
+  gap: 'text-to-visual',
+  height: {
+    density: {
+      compact: 32,
+      regular: 48
+    }
+  },
+  alignItems: 'center',
+  position: 'relative',
+  cursor: 'default',
   flexShrink: 0,
-  fontFamily: 'sans',
-  fontWeight: 'normal',
-  flexDirection: {
-    orientation: {
-      horizontal: 'column'
+  transition: 'default',
+  paddingX: {
+    labelBehavior: {
+      hide: size(6)
     }
   }
-}, getAllowedOverrides({height: true}));
+}, getAllowedOverrides());
 
-/**
- * Tabs organize content into multiple sections and allow users to navigate between them. The content under the set of tabs should be related and form a coherent unit.
- */
-export const Tabs = forwardRef(function Tabs(props: TabsProps, ref: DOMRef<HTMLDivElement>) {
-  [props, ref] = useSpectrumContextProps(props, ref, TabsContext);
-  let {
-    density = 'regular',
-    isDisabled,
-    disabledKeys,
-    orientation = 'horizontal'
-  } = props;
-  let domRef = useDOMRef(ref);
+const icon = style({
+  display: 'block',
+  flexShrink: 0,
+  '--iconPrimary': {
+    type: 'fill',
+    value: 'currentColor'
+  }
+});
+
+export function Tab(props: TabProps) {
+  let {density, labelBehavior} = useContext(InternalTabsContext) ?? {};
 
   return (
-    <RACTabs
+    <RACTab
       {...props}
-      ref={domRef}
+      // @ts-ignore
+      originalProps={props}
       style={props.UNSAFE_style}
-      className={renderProps => (props.UNSAFE_className || '') + tabs({...renderProps}, props.styles)}>
-      <Provider
-        values={[
-          [TabsContext, {density, isDisabled, disabledKeys, orientation}]
-        ]}>
+      className={renderProps => (props.UNSAFE_className || '') + tab({...renderProps, density, labelBehavior}, props.styles)}>
+      {({
+          // @ts-ignore
+          isMenu
+        }) => {
+        if (isMenu) {
+          return props.children;
+        } else {
+          return (
+            <Provider
+              values={[
+                [TextContext, {
+                  styles:
+                    style({
+                      order: 1,
+                      display: {
+                        labelBehavior: {
+                          hide: 'none'
+                        }
+                      }
+                    })({labelBehavior})
+                }],
+                [IconContext, {
+                  render: centerBaseline({slot: 'icon', styles: style({order: 0})}),
+                  styles: icon
+                }]
+              ]}>
+              {typeof props.children === 'string' ? <Text>{props.children}</Text> : props.children}
+            </Provider>
+          );
+        }
+      }}
+    </RACTab>
+  );
+}
+
+const tabPanel = style({
+  display: 'flex',
+  marginTop: 4,
+  marginX: -4,
```

Comment:
i adjusted some styles in here to make it easier to create a panel that scrolls for overflow and compensates for our focus halos automatically

it also automatically lays out in such a way that users can place a child which will stretch to the fill the space because it's now default display: flex

let me know if this is too controversial

another option was to use allowedOverrides for styles and add display/padding to it